File: e2e-tests/playwright/java/src/test/java/net/gradido/e2e/components/Toasts.java
package net.gradido.e2e.components;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;

import static org.junit.jupiter.api.Assertions.assertTrue;

public class Toasts {
    public final Locator toastSlot;
    public final Locator toastTypeError;
    public final Locator toastTitle;
    public final Locator toastMessage;

    public Toasts(Page page) {
        toastSlot = page.locator("#__BVID__toaster-container");
        toastTypeError = toastSlot.locator(".toast.text-bg-danger");
        toastTitle = toastTypeError.locator(".gdd-toaster-title");
        toastMessage = toastTypeError.locator(".gdd-toaster-body");
    }

    public void assertErrorToastVisible() {
        toastTypeError.waitFor(); 
        assertTrue(toastTitle.isVisible());
        assertTrue(toastMessage.isVisible());
    }
}

File: e2e-tests/playwright/typescript/components/Toasts.ts
import type { Locator, Page } from '@playwright/test'
import { expect } from '@playwright/test'

export class Toasts {
  public readonly toastSlot: Locator
  public readonly toastTypeError: Locator
  public readonly toastTitle: Locator
  public readonly toastMessage: Locator

  constructor(page: Page) {
    this.toastSlot = page.locator('#__BVID__toaster-container')
    this.toastTypeError = this.toastSlot.locator('.toast.text-bg-danger')
    this.toastTitle = this.toastTypeError.locator('.gdd-toaster-title')
    this.toastMessage = this.toastTypeError.locator('.gdd-toaster-body')
  }

  async assertErrorToastVisible(): Promise<void> {
    await this.toastTypeError.waitFor({ state: 'visible' })
    expect(this.toastTitle).toBeVisible()
    expect(this.toastMessage).toBeVisible()
  }
}
